internal/storage/pmem/pmem.go

```diff
@@ -25,15 +27,19 @@ var (
 const (
 	pMemFmt         = "/dev/pmem%d"
 	linearDeviceFmt = "dm-linear-pmem%d-%d-%d"
+	verityDeviceFmt = "dm-verity-pmem%d-%s"
 )
 
+// mountInternal mounts source to target via unix.Mount
 func mountInternal(source, target string) (err error) {
 	if err := osMkdirAll(target, 0700); err != nil {
 		return err
 	}
 	defer func() {
 		if err != nil {
-			osRemoveAll(target)
+			if err := osRemoveAll(target); err != nil {
+				log.G(context.TODO()).WithError(err).Debugf("error cleaning up target: %s", target)
```

Comment:
We could add a `ctx context.Context` parameter to `mountInternal` so we don't have to use a fresh context and just pass in the one we get from `Mount`